R/fread.R

```diff
@@ -81,6 +90,59 @@ yaml=FALSE, tmpdir=tempdir(), tz="UTC")
     }
     file = tmpFile
   }
+  connection_spill_info = NULL
+  if (input_is_con) {
+    if (verbose) {
+      catf("[00] Spill connection to tempfile\n  Connection class: %s\n  Reading connection into RAM buffer... ", toString(class(input)))
+      flush.console()
+    }
+    spill_started.at = proc.time()
+    con_summary = summary(input)
+    con_desc = con_summary$description
+    con_class = class1(input)
+    con_open = isOpen(input)
+
+    needs_reopen = FALSE
+    if (con_open) {
+      binary_modes = c("rb", "r+b", "wb", "w+b", "ab", "a+b")
+      if (!con_summary$mode %chin% binary_modes) needs_reopen = TRUE
+    }
+
+    close_con = NULL
+
+    if (needs_reopen) {
+      close(input)
+      input = switch(con_class,
```

Comment:
Let's make this an S3 method with the below as exported methods, then other downstream connection types can easily define methods that "just work" on other custom connections, WDYT?

Reply:
done, see below.